lib/resources/iis_site.rb

```diff
@@ -107,12 +117,17 @@ def iis_site(name)
         "#{k['protocol']} #{k['bindingInformation']}#{k['protocol'] == 'https' ? " sslFlags=#{k['sslFlags']}" : ''}"
       }
 
+      bindings_array_with_cert_hash = site['bindings']['Collection'].map { |k|
```

Comment:
Hrm.  'bindings_array' is really regrettable (and not your doing). It collapses a rich datastructure into a string, which people are now matching against. The field selection, formatting, and order now must never change lest they break other tests.  The fact that there are multiple bindings just makes things worse; the implementation of `has_binding?` is very brittle.

Which puts you, @robwafle, in an tough position - to add a field, it looks like the only thing to do might be to add a new property, which also stringifies, but with your new field added.  When someone else wants another field, they'd have to add it as a property too, but we'd also have to consider combinations....

I'm no IIS expert, but I think we could have better matchers here.  We could do this: 
 * Store the entire, un-stringified binding information in the `info` hash
 * implement a new matcher, `has_binding_with?(criteria = {})`

Then, `has_binding_with` could accept criteria, and you would return true if there was at lease one binding that met all the criteria.

Again, I'm no IIS expert, but from the examples it would be easy to see:
```
describe iis_site(...) do
  it { should have_binding_with(schema: :https, port: 443, hostname: 'www.contoso.com') }
  it { should have_binding_with(certificate_hash: '123456789sdfasdf', port: 443) }
end
```

This would allow people to test the cert hash if they wanted to.  We could add new criteria in the future without breaking existing users' code.  

What do you think?

Reply:
Yes.  I couldn't agree more.  My initial change is functional and was able to solve my immediate problem, but what you proposed is without a doubt leaps and bounds better.  I discussed this with another gentleman on the chef community slack #inspec channel and we were thinking the same.  I will update this PR to add the have_binding_with() method as above and remove the has_binding_with_cert() method.

Reply:
Glad we got this worked out :) 👍